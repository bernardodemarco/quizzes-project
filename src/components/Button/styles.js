import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme, background, fontSize, fontWeight, lineHeight }) => css`
    background-color: ${background ? theme.colors.primary : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.buttons[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};

    font-family: inherit;
    width: 100%;
    height: 5.6rem;
    text-align: center;
    border: none;
    border-radius: 1.6rem;
    padding: 1.6rem 3.2rem;

    &:hover {
      cursor: pointer;
    }
  `}
`;
