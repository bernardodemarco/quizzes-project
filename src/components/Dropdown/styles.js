import styled, { css } from 'styled-components';

export const Dropdown = styled.ul`
  ${({ theme, isVisible }) => css`
    visibility: hidden;
    opacity: 0;

    transition: all 300ms ease-in-out;

    ${
      isVisible &&
      css`
      visibility: visible;
      opacity: 1;
    `
    }

    position: absolute;
    top: 10rem;
    right: 38%;

    width: 14.5rem;
    height: 15.2rem;

    background-color: ${theme.colors.white};
    border-radius: 0.8rem;
    box-shadow: 0rem 0.4rem 0.4rem rgba(0, 0, 0, 0.25);

    list-style: none;

    @media (max-width: 785px) {
      visibility: hidden;
      opacity: 0;
    }
  `}
`;

export const DropdownItem = styled.li`
  ${({ theme }) => css`
    width: 100%;

    padding: 1rem 4.5rem;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    &:last-child {
      border-bottom: none;
    }

    &:hover {
      cursor: pointer;
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: #000000;

    text-align: center;

    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    font-weight: ${theme.fonts.fontWeights.normal};
    line-height: 1.8rem;
  `}
`;
