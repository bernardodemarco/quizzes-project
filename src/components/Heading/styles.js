import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme, fontSize, fontWeight, lineHeight }) => css`
    color: ${theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.headings[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};
  `}
`;
