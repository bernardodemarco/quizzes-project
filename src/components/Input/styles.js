import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 1.2rem;

    width: 100%;
    height: 5.3rem;
    padding: 1.6rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 75%;
    border: none;
    font-family: inherit;
    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    color: ${theme.colors.darkGray};
  `}
`;

export const Icon = styled.img`
  ${({ hover = false }) => css`
    width: 3rem;
    height: 2.5rem;

    &:hover {
      cursor: ${hover ? 'pointer' : 'default'};
    }
  `}
`;
