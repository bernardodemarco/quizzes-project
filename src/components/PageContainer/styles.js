import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    width: 100%;
    max-width: 114.6rem;
    min-height: 100vh;

    margin: 0 auto;
    overflow-x: hidden;

    display: flex;
    align-items: center;
  `}
`;
