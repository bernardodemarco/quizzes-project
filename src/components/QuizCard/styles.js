import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    overflow: hidden;

    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 0.8rem;
    filter: drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.25));

    width: 55.2rem;
    height: fit-content;
    background-color: ${theme.colors.white};

    &:hover {
      cursor: pointer;
    }

    @media ${theme.media.largeDevices} {
      width: 45rem;
    }

    @media ${theme.media.mediumDevices} {
      width: 55.2rem;
    }

    @media ${theme.media.smallDevices} {
      width: 90vw;

      p {
        text-align: start;
      }
    }
  `}
`;

export const Image = styled.div`
  ${({ imgSrc }) => css`
    // position set to relative in order to position <QuizDifficulty /> accordingly
    position: relative;

    width: 100%;
    height: 19.4rem;

    background-image: url(${imgSrc});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  `}
`;

export const TextContainer = styled.div`
  ${() => css`
    width: 100%;
    padding: 1.6rem 1.6rem 0.8rem 1.6rem;

    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 0.4rem;
  `}
`;

export const FeedbackContainer = styled.div`
  ${() => css`
    width: 100%;

    display: flex;
    justify-content: space-between;
  `}
`;
