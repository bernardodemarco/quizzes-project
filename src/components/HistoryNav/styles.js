import styled, { css } from 'styled-components';

import { Container as PageContainer } from './../PageContainer/styles';
import { ImageContainer as ReturnButton } from './../ReturnButton/styles';
import { Heading } from './../Heading/styles';

export const Nav = styled.nav`
  ${({ theme }) => css`
    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;
    left: 0;

    height: 12rem;
    border-bottom: 0.1rem solid ${theme.colors.gray};

    ${PageContainer} {
      min-height: initial;
      height: 100%;
      justify-content: flex-start;
      align-items: center;
    }

    ${Heading} {
      z-index: 1;
      position: fixed;
      left: 14.7rem;
    }

    @media ${theme.media.mediumDevices} {
      ${ReturnButton} {
        left: 5rem;
      }
    }

    @media ${theme.media.smallDevices} {
      ${Heading} {
        left: 12rem;
        font-size: ${theme.fonts.fontSizes.headings.medium};
      }
    }
  `}
`;
