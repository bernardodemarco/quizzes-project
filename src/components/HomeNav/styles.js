import styled, { css } from 'styled-components';

import { Container as PageContainer } from '../PageContainer/styles';
import { Input, Container as InputContainer } from '../Input/styles';
import { Heading } from '../Heading/styles';

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;

    z-index: 2;
    position: fixed;
    top: 3rem;
    right: 3rem;

    height: 4rem;
    width: 4rem;

    color: ${theme.colors.primary};
    background-color: ${theme.colors.lightGray};
    border: none;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      display: block;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    height: 12rem;

    z-index: 1;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    transition: all 300ms ease-in-out;

    & ${PageContainer} {
      min-height: initial;
      height: 100%;
    }

    @media (max-width: 1300px) {
      & ${PageContainer} {
        padding: 0 3rem;
      }
    }

    @media (max-width: 785px) {
      & {
        height: 100vh;
        bottom: 0;
        border-bottom: none;

        visibility: hidden;
        opacity: 0;
        ${
          isVisible &&
          css`
          visibility: visible;
          opacity: 1;
        `
        }
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;

    @media (max-width: 785px) {
      & {
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 7.5rem;
      }
    }
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    & ${Heading} {
      text-align: center;
    }

    @media (max-width: 1130px) {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.medium};
        margin-right: 2rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.small};
      }
    }

    @media (max-width: 785px) {
      & {
        justify-content: center;
        align-items: center;
      }

      & ${Heading} {
        margin-right: 0;
      }
    }
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 3rem;

    & ${Input} {
      width: 40.4rem;
    }

    @media (max-width: 1130px) {
      & {
        align-items: center;
      }

      & ${Input} {
        width: 30rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Input} {
        width: 22.5rem;
      }
    }

    @media (max-width: 785px) {
      & {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & ${InputContainer}, & ${Input}, & .dropdown-button  {
        display: none;
      }
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    text-align: center;
    color: ${theme.colors.dark};

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        font-size: 1.8rem;
      }
    }
  `}
`;

export const Theme = styled.p`
  ${({ theme }) => css`
    display: none;
    color: ${theme.colors.dark};
    text-align: center;

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        display: block;
        font-size: 1.8rem;
      }
    }
  `}
`;
