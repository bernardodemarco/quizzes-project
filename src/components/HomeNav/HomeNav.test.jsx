import { fireEvent, screen } from '@testing-library/react';

import { HomeNav } from '.';
import { BrowserRouter as Router } from 'react-router-dom';
import { renderThemeProvider } from './../../styles/render-theme-provider';

const username = 'username';

describe('<HomeNav />', () => {
  it('should render <HomeNav /> correctly and when clicking in "Temas", <Dropdown /> should appear', () => {
    renderThemeProvider(
      <Router>
        <HomeNav username={username} />
      </Router>,
    );

    expect(screen.getByRole('heading', { name: username })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'main menu' }),
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Pesquisar quiz')).toBeInTheDocument();
    expect(screen.getByLabelText('Open/Close menu')).toHaveStyle({
      display: 'none',
    });

    expect(screen.queryByRole('list')).not.toBeInTheDocument();
    const dropdownBtn = screen.getByText('Temas');

    fireEvent.click(dropdownBtn);

    expect(screen.getByRole('list')).toHaveStyle({
      opacity: 1,
      visibility: 'visible',
    });

    fireEvent.click(screen.getByRole('list'));

    expect(screen.queryByRole('list')).not.toBeInTheDocument();
  });

  it('should allow dropdown items to be clicked', () => {
    const func = jest.fn();
    renderThemeProvider(
      <Router>
        <HomeNav username={username} onDropdownItemClick={func} />
      </Router>,
    );

    fireEvent.click(screen.getByText('Temas'));

    const list = screen.getByRole('list');

    fireEvent.click(list.firstChild.firstChild);
  });

  it('when using mobile devices, it should apply the styles accordingly', () => {
    renderThemeProvider(
      <Router>
        <HomeNav username={username} />
      </Router>,
    );

    const button = screen.getByLabelText('Open/Close menu');
    expect(button).toHaveStyleRule('display', 'block', {
      media: '(max-width: 785px)',
    });

    const menuContainer = button.nextSibling;
    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: '(max-width: 785px)',
    });

    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);

    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: '(max-width: 785px)',
    });
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();

    fireEvent.click(screen.getByRole('navigation'));

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: '(max-width: 785px)',
    });
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderThemeProvider(
      <Router>
        <HomeNav username={username} />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
