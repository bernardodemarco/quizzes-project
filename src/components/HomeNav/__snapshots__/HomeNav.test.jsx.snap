// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeNav /> should match snapshot 1`] = `
.c4 {
  width: 100%;
  max-width: 125rem;
  min-height: 100vh;
  margin: 0 auto;
  overflow-x: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c17 {
  border: 0.1rem solid #BEBAB3;
  border-radius: 1.2rem;
  width: 100%;
  height: 5.3rem;
  padding: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c19 {
  width: 75%;
  border: none;
  font-family: inherit;
  font-weight: 400;
  font-size: 1.6rem;
  color: #78746D;
}

.c20 {
  width: 3rem;
  height: 2.5rem;
}

.c20:hover {
  cursor: default;
}

.c9 {
  color: #3C3A36;
  font-size: 4rem;
  font-weight: 700;
  line-height: 4.2rem;
}

.c0 {
  display: none;
  z-index: 2;
  position: fixed;
  top: 3rem;
  right: 3rem;
  height: 4rem;
  width: 4rem;
  color: #82327E;
  background-color: #FFF5EE;
  border: none;
}

.c0:hover {
  cursor: pointer;
}

.c2 {
  height: 12rem;
  z-index: 3;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background-color: #FFFFFF;
  border-bottom: 0.1rem solid #BEBAB3;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c2 .c3 {
  min-height: initial;
  height: 100%;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c6 .c8 {
  text-align: center;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  gap: 3rem;
}

.c10 .c18 {
  width: 40.4rem;
}

.c14 {
  display: block;
  text-align: center;
  color: #3C3A36;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.1rem;
}

.c14:hover {
  cursor: pointer;
}

.c15 {
  display: none;
  color: #3C3A36;
  text-align: center;
  font-weight: 400;
  font-size: 1.6rem;
  line-height: 2.1rem;
}

.c15:hover {
  cursor: pointer;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  font-size: 1.6rem;
  font-weight: 400;
  color: #3C3A36;
  text-align: center;
  line-height: 2.6rem;
}

.c11 {
  visibility: hidden;
  opacity: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  position: absolute;
  top: 10rem;
  right: 38%;
  width: 14.5rem;
  height: 15.2rem;
  background-color: #FFFFFF;
  border-radius: 0.8rem;
  box-shadow: 0rem 0.4rem 0.4rem rgba(0,0,0,0.25);
  list-style: none;
}

.c12 {
  width: 100%;
  padding: 1rem 4.5rem;
  border-bottom: 0.1rem solid #BEBAB3;
}

.c12:last-child {
  border-bottom: none;
}

.c12:hover {
  cursor: pointer;
}

.c13 {
  color: #000000;
  text-align: center;
  font-size: 1.6rem;
  font-weight: 500;
  line-height: 1.8rem;
}

@media (max-width:785px) {
  .c0 {
    display: block;
  }
}

@media (max-width:1300px) {
  .c2 .c3 {
    padding: 0 3rem;
  }
}

@media (max-width:785px) {
  .c2 {
    height: 100vh;
    bottom: 0;
    border-bottom: none;
    visibility: hidden;
    opacity: 0;
  }
}

@media (max-width:785px) {
  .c5 {
    height: 100%;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    gap: 7.5rem;
  }
}

@media (max-width:1130px) {
  .c6 .c8 {
    font-size: 3.2rem;
    margin-right: 2rem;
  }
}

@media (max-width:974px) {
  .c6 .c8 {
    font-size: 2.4rem;
  }
}

@media (max-width:785px) {
  .c6 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c6 .c8 {
    margin-right: 0;
  }
}

@media (max-width:1130px) {
  .c10 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c10 .c18 {
    width: 30rem;
  }
}

@media (max-width:974px) {
  .c10 .c18 {
    width: 22.5rem;
  }
}

@media (max-width:785px) {
  .c10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c10 .c16,
  .c10 .c18,
  .c10 .dropdown-button {
    display: none;
  }
}

@media (max-width:785px) {
  .c14 {
    font-size: 1.8rem;
  }
}

@media (max-width:785px) {
  .c15 {
    display: block;
    font-size: 1.8rem;
  }
}

@media (max-width:785px) {
  .c11 {
    visibility: hidden;
    opacity: 0;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        >
          <p
            class="c7"
            color="dark"
            font-size="large"
            font-weight="light"
          >
            Olá,
          </p>
          <h1
            class="c8 c9"
            font-size="large"
            font-weight="bold"
          >
            username
          </h1>
        </div>
        <nav
          aria-label="main menu"
          class="c10"
        >
          <ul
            class="c11"
          >
            <li
              class="c12"
            >
              <p
                class="c13"
              >
                #HTML
              </p>
            </li>
            <li
              class="c12"
            >
              <p
                class="c13"
              >
                #UX
              </p>
            </li>
            <li
              class="c12"
            >
              <p
                class="c13"
              >
                #Swift
              </p>
            </li>
            <li
              class="c12"
            >
              <p
                class="c13"
              >
                #UI
              </p>
            </li>
          </ul>
          <a
            class="c14"
          >
            Histórico
          </a>
          <span
            class="c14 dropdown-button"
          >
            Temas
          </span>
          <p
            class="c15 mobile-quiz-search"
          >
            #HTML
          </p>
          <p
            class="c15 mobile-quiz-search"
          >
            #UX
          </p>
          <p
            class="c15 mobile-quiz-search"
          >
            #Swift
          </p>
          <p
            class="c15 mobile-quiz-search"
          >
            #UI
          </p>
          <div
            class="c16 c17"
          >
            <input
              class="c18 c19"
              placeholder="Pesquisar quiz"
              required=""
              type="search"
            />
            <img
              alt="search input icon"
              class="c20"
              src="assets/images/search-icon.svg"
            />
          </div>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
